mzLib/Predictions/Koina/SupportedModels/Prosit2020IntensityHCD/Prosit2020IntensityHCD.cs

```diff
@@ -0,0 +1,332 @@
+        public List<int> PrecursorCharges = new();
+        public List<int> CollisionEnergies = new();
+        public List<double?> RetentionTimes = new();
+        public List<LibrarySpectrum> PredictedSpectra = new();
+        public double MinIntensityFilter; // Tolerance for intensity filtering of predicted peaks
+
+
+        public Prosit2020IntensityHCD(List<string> peptideSequences, List<int> precursorCharges, List<int> collisionEnergies, List<double?> retentionTimes, out WarningException? warnings, double? minIntensityFilter=null)
+        {
+            // Verify input lists are of the same length
+            if (peptideSequences.Count != precursorCharges.Count
+                || precursorCharges.Count != collisionEnergies.Count
+                || collisionEnergies.Count != retentionTimes.Count)
+            {
+                throw new ArgumentException("Input lists must have the same length.");
+            }
+
+            // Ensure that the inputs meet the model requirements.
+            List<string> invalidArguments = new();
+            for (int i = 0; i < peptideSequences.Count; i++)
+            {
+                var peptide = peptideSequences[i];
+                var charge = precursorCharges[i];
+                var energy = collisionEnergies[i];
+                var retentionTime = retentionTimes[i];
+
+                // Skip invalid peptides
+                if (!HasValidModifications(peptide) ||
+                    !IsValidSequence(peptide) ||
+                    !AllowedPrecursorCharges.Contains(charge) ||
+                    energy <= 0
+                    )
+                {
+                    invalidArguments.Add($"Index {i}: Peptide '{peptide}' (Length: {peptide.Length}), Charge: {charge}, Collision Energy: {energy}");
+                }
+                else
+                {
+                    PeptideSequences.Add(ConvertToPrositModificationFormat(peptide));
+                    PrecursorCharges.Add(charge);
+                    CollisionEnergies.Add(energy);
+                    RetentionTimes.Add(retentionTime);
+                }
+            }
+
+            warnings = null;
+            if (invalidArguments.Count > 0)
+            {
+                string warningMessage = "The following input entries are invalid and will be skipped:\n"
+                    + string.Join("\n", invalidArguments)
+                    + "\nModel Requirements:\n"
+                    + $"- Peptide length <= {MaxPeptideLength}\n"
+                    + "- Peptide sequence is not empty\n"
+                    + "- Peptide has valid modifications\n"
+                    + $"- Precursor charge in [{string.Join(", ", AllowedPrecursorCharges)}]\n"
+                    + "- Collision energy > 0\n"
+                    + "- Retention time is null or >= 0";
+                warnings = new WarningException(warningMessage);
+            }
+
+            MinIntensityFilter = minIntensityFilter ?? 1e-4;
```

Comment:
You could add a default minimum intensity as a constant if you anticipated the same one will be used when the other constructors are implemented